specification/ml/_types/TrainedModel.ts

```diff
@@ -333,31 +350,39 @@ export class TrainedModelAllocationRoutingTable {
    * The current routing state.
    */
   routing_state: RoutingState
+  /**
+   * Current number of allocations.
+   */
+  current_allocations: integer
+  /**
+   * Target number of allocations.
+   */
+  target_allocations: integer
 }
 
-export class TrainedModelDeploymentAllocationStatus {
+export class TrainedModelDeploymentAssignmentStatus {
   /** The current number of nodes where the model is allocated. */
   allocation_count: integer
   /** The detailed allocation state related to the nodes. */
-  state: DeploymentAllocationState
+  state: DeploymentAssignmentState
   /** The desired number of nodes for model allocation. */
   target_allocation_count: integer
 }
 
-export class TrainedModelAllocation {
+export class TrainedModelAssignment {
   /**
-   * The overall allocation state.
+   * The overall assignment state.
    */
-  allocation_state: DeploymentAllocationState
+  assignment_state: DeploymentAssignmentState
```

Comment:
This should be a new object. Assignment state is different than the allocation state (which we are keeping)

This is assignment state, which is an enumeration of: "starting", "started", "stopping", "failed".